app/views/api/v2/layouts/permissions.json.rabl

```diff
@@ -0,0 +1,8 @@
+object @resource
+
+if params.has_key?(:include_permissions)
+  # To avoid loading all the records into the cache
+  # authorized_for is used instead of authorizer.can?.
+  node(:can_edit) { |resource| authorized_for(:action => :edit, :auth_object => resource, :authorizer => authorizer, :permission => "edit_#{controller_name.split('/').last}") }
```

Comment:
Since you now specify permission explicitly and no longer pass controller, the action can be dropped (also on line below). But before you do so, why we need to pass permission explicitly? Was it unable to detect the right permission inside `authorized_for`? It seems you're stripping the controller namespace, which in fact might be useful for other `authorized_for` calls, so perhaps fix it there? Then we'd be only sending the action right?

If you feel this is unrelated to this PR, we can fix just this separately and rebase this PR on top.

Reply:
the permission that is set in `authorized_for` is `{edit,destroy}_api/v2/models` which tells if the user has permissions to edit or delete the resource through the API while I would like it to tell me if the user has permissions to edit/delete the resource through the web which for that I need to pass `permission` to `authorized_for`.

I am using the `controller_name` to convert it to `edit/destroy_{resource_name}` from `api/v2/{resource_name}`.

Reply:
Ah, now I see why you added that, thanks. I think you could then perhaps reuse [controller_permission](https://github.com/theforeman/foreman/blob/develop/app/controllers/api/base_controller.rb#L53-L55) which should always be set correctly without the namespace. You might need to add `helper_method :controller_name` to the base controller so it's usable in views.

And that means, the `:action => :edit` can be dropped now. The `authorized_for` needs this only to build permission name if auth_object is set, but since we pass that explicitly, it's not used anymore.

Then I think we're good in terms permissions verification.